go/ct/rlz/conditions.go

```diff
@@ -655,3 +655,88 @@ func (c *isAddressCold) GetTestValues() []TestValue {
 func (c *isAddressCold) String() string {
 	return fmt.Sprintf("cold(%v)", c.key)
 }
+
+////////////////////////////////////////////////////////////
+// Has Address Selfdestructed
+
+type hasSelfDestructed struct {
+	isSet bool
+}
+
+func HasSelfDestructed() Condition {
+	return &hasSelfDestructed{true}
+}
+
+func (c *hasSelfDestructed) Check(s *st.State) (bool, error) {
+	if c.isSet {
+		return s.HasSelfDestructed, nil
+	}
+	return true, nil
+
+}
+
+func (c *hasSelfDestructed) Restrict(generator *gen.StateGenerator) {
+	generator.SelfDestruct()
+}
+
+func (c *hasSelfDestructed) GetTestValues() []TestValue {
+	property := Property(c.String())
+	domain := boolDomain{}
+	restrict := func(generator *gen.StateGenerator, hasSelfDestructed bool) {
+		if hasSelfDestructed {
+			generator.SelfDestruct()
+		} else {
+			generator.NotSelfDestruct()
+		}
+	}
+	return []TestValue{
+		NewTestValue(property, domain, true, restrict),
+		NewTestValue(property, domain, false, restrict),
+	}
+}
+
+func (c *hasSelfDestructed) String() string {
+	return fmt.Sprintf("hasSelfDestructed(%v)", c.isSet)
+}
+
+////////////////////////////////////////////////////////////
+// Has Not Address Selfdestructed
+
+type hasNotSelfDestructed struct {
+	isSet bool
+}
+
+func HasNotSelfDestructed() Condition {
+	return &hasNotSelfDestructed{true}
+}
+
+func (c *hasNotSelfDestructed) Check(s *st.State) (bool, error) {
+	if c.isSet {
+		return !s.HasSelfDestructed, nil
+	}
+	return true, nil
+}
+
+func (c *hasNotSelfDestructed) Restrict(generator *gen.StateGenerator) {
+	generator.NotSelfDestruct()
+}
+
+func (c *hasNotSelfDestructed) GetTestValues() []TestValue {
+	property := Property(c.String())
+	domain := boolDomain{}
+	restrict := func(generator *gen.StateGenerator, hasSelfDestructed bool) {
+		if hasSelfDestructed {
+			generator.SelfDestruct()
+		} else {
+			generator.NotSelfDestruct()
+		}
+	}
+	return []TestValue{
+		NewTestValue(property, domain, true, restrict),
+		NewTestValue(property, domain, false, restrict),
+	}
```

Comment:
Since it is the negation of `hasSelfDestructed` you should just call
```suggestion
	return HasSelfDestructed().GetTestValues()
```
This also solves the problem that those two conditions should target the same property.